Helper/Order.php

```diff
@@ -762,42 +762,30 @@ public function saveUpdateOrder($reference, $storeId = null, $boltTraceId = null
     }
 
     /**
-     * @param \Magento\Quote\Model\Quote $quote
-     * @param string                     $transaction
-     * @return OrderModel
-     * @throws BoltException
-     * @throws LocalizedException
+     * Try to fetch and price-validate already existing order.
+     * Use case: order has been created but the payment failes due to the ivnalid credit card data.
+     * Then the customer enters the correct card info, the previously created order is used if the amounts match.
+     *
+     * @param Quote $quote
+     * @param \stdClass $transaction
+     * @return bool|false|OrderModel
+     * @throws \Exception
      */
-    public function preAuthCreateOrder($quote, $transaction)
+    public function processExistingOrder($quote, $transaction)
     {
         // check if the order has been created in the meanwhile
         if ($order = $this->getExistingOrder($quote->getReservedOrderId())) {
             if ($this->hasSamePrice($order, $transaction)) {
                 return $order;
-            } else {
-                $this->deleteOrder($order);
             }
```

Comment:
I want to suggest add here an event which will help a merchant's system to avoid suddenly disappearing data.
I am talking about some ERP or Reporting system which collect order data. It is rarely but happened.

Reply:
As i see our method `deleteOrder()` make cancel at first and then `delete`.
As alternative we can add event directly to the method `deleteOrder()`.